### PR TITLE
[CI:DOCS] Remove broken CIRRUS_BASE_SHA usage

### PR DESCRIPTION
**Pull request by cevich**
Unfortunately this value doesn't properly reflect the current branch point of a PR.  Replace it with a call to `git merge-base` instead.

Ref: https://github.com/cirruslabs/cirrus-ci-docs issue number 1279

### CODE REVIEW

**Comment by cevich** [comment]
Manually tested with this PR rebased on [an older commit from main + no IMG_SFX update](https://cirrus-ci.com/task/6205488717627392).

**Comment by cevich** [comment]
Manually tested with [an already used IMG_SFX](https://cirrus-ci.com/task/4657716319223808).

**Comment by cevich** [comment]
@edsantiago when you get a chance, this one isn't critical.  Lokesh ran into the broken behavior today in his PR because he hadn't rebased.  This should fix that for everybody.

File: ci/validate.sh [hunk 1]
```diff
@@ -52,17 +52,20 @@ if [[ "$CIRRUS_PR_TITLE" =~ CI:DOCS ]]; then
...
-# Variable is defined by Cirrus-CI at runtime
-# shellcheck disable=SC2154
-if ! git diff --name-only ${CIRRUS_BASE_SHA}..HEAD | grep -q IMG_SFX; then
+# Fix "Not a valid object name main" error from Cirrus's
+# incomplete checkout.
+git remote update origin
+# Determine where PR branched off of $CIRRUS_BASE_BRANCH
+base_sha=$(git merge-base origin/${CIRRUS_BASE_BRANCH:-main} HEAD)
 
+# Variable is defined by Cirrus-CI at runtime
```
Comment (edsantiago) [inline ci/validate.sh:61]: Comment is incorrect

File: ci/validate.sh [hunk 2]
```diff
@@ -52,17 +52,20 @@ if [[ "$CIRRUS_PR_TITLE" =~ CI:DOCS ]]; then
...
 fi
 
-# Variable is defined by Cirrus-CI at runtime
-# shellcheck disable=SC2154
-if ! git diff --name-only ${CIRRUS_BASE_SHA}..HEAD | grep -q IMG_SFX; then
+# Fix "Not a valid object name main" error from Cirrus's
+# incomplete checkout.
+git remote update origin
+# Determine where PR branched off of $CIRRUS_BASE_BRANCH
+base_sha=$(git merge-base origin/${CIRRUS_BASE_BRANCH:-main} HEAD)
```
Comment (edsantiago) [inline ci/validate.sh:59]: I remember trying `origin/` in my podman PR, and it failed so I removed it. I can't remember what the failure was. Do you remember why you added `origin/` ?

**Review by edsantiago — COMMENTED** [review]
LGTM but one question and one misleading comment

File: ci/validate.sh (hunk 2 above)
Comment (cevich) [reply ci/validate.sh:59]: Yes, it's in the comment above the `git remote update origin` :wink:

File: ci/validate.sh (hunk 1 above)
Comment (cevich) [reply ci/validate.sh:61]: Oh woops, good catch.  Thanks.

**Comment by edsantiago** [comment]
What bothers me is that you added `origin` **after** all your testing. So `origin/` hasn't actually been tested, because it hasn't even been run once. Before merging this, could you retry your tests?

**Comment by cevich** [comment]
> after all your testing

I was doing a lot of re-run in terminal to figure that `origin/` thing out, I thought I did run tests with it :thinking: but I'll run them again regardless...

**Comment by cevich** [comment]
...actually, no.  For sure both tests had the `origin/` line in them.

- Manually tested with this PR rebased on [an older commit from main + no IMG_SFX update](https://cirrus-ci.com/task/6205488717627392) - Uses b331fe4 having the [origin](https://github.com/containers/automation_images/commit/b331fe43e1616f7cf4d90075e13ed3df7f928a2e#diff-e1944be4ef28c744c8b6f698adbd32ef54a0214576b272b59db7fe2081674c32R60) prefix.
- Manually tested with [an already used IMG_SFX](https://cirrus-ci.com/task/4657716319223808) - Uses 8228459 (a DO NOT MERGE), with parent b331fe4 having the [origin](https://github.com/containers/automation_images/commit/b331fe43e1616f7cf4d90075e13ed3df7f928a2e#diff-e1944be4ef28c744c8b6f698adbd32ef54a0214576b272b59db7fe2081674c32R60) prefix.

**Comment by edsantiago** [comment]
Hmm, you're right. The github web UI `Compare` link is showing misleading diffs. Sorry!

LGTM

**Pull request merged**